packages/dialtone-vue2/recipes/conversation_view/wysiwyg_editor/wysiwyg_editor.vue

```diff
@@ -0,0 +1,574 @@
+<!-- eslint-disable vue/no-restricted-class -->
+<template>
+  <div
+    data-qa="dt-wysiwyg-editor"
+    role="presentation"
+    :class="['d-d-flex', 'd-fd-column', 'd-baw1', 'd-ba', {
+      'd-bc-black-500': hasFocus,
+      'd-bc-default': !hasFocus,
+      'd-bar8': roundedEdges,
+      'd-bar0': !roundedEdges,
+    }]"
+    @click="$refs.richTextEditor.focusEditor()"
+  >
+    <!-- Section for the top UI -->
+    <div
+      class="d-d-flex d-fd-row d-px8 d-py2 d-bgc-black-200"
+      :class="{ 'd-btr8': roundedEdges, 'd-btr0': !roundedEdges }"
+    >
+      <dt-button
+        v-if="showBoldButton"
+        data-qa="dt-wysiwyg-editor-bold-btn"
+        class="h:d-bgc-black-300 d-fc-black-700 d-mx4"
+        :class="{ 'd-bgc-black-300 d-fc-black-900': $refs.richTextEditor?.editor?.isActive('bold') }"
+        importance="clear"
+        kind="inverted"
+        size="sm"
+        @click="onBoldTextToggle"
+      >
+        <template #icon>
+          <dt-icon
+            name="bold"
+            size="200"
+            class="d-fw-bold"
+          />
+        </template>
+      </dt-button>
+
+      <dt-button
+        v-if="showItalicsButton"
+        data-qa="dt-wysiwyg-editor-italics-btn"
+        class="h:d-bgc-black-300 d-fc-black-700 d-mx4"
+        :class="{ 'd-bgc-black-300 d-fc-black-900': $refs.richTextEditor?.editor?.isActive('italic') }"
```

Comment:
I'm not sure if I maybe I'm misunderstanding how to use these button props, but I think the problem here is we want to set these styles to simulate the button being 'pressed' or 'hovered on' for when you place the cursor on a text with an applied style. Also there's one more issue which is that the 'muted' hover style matches the color of the background color of the tools bar (at least in light mode), so I needed a slightly darker shade of gray.

<img width="670" alt="Screenshot 2024-01-23 at 10 51 48" src="https://github.com/dialpad/dialtone/assets/12575023/abeb01b0-236c-4855-b42f-b170bd161d38">

Maybe we can achieve this with the kind/importance props, but I haven't able to get it right so far...

Reply:
> we want to set these styles to simulate the button being 'pressed' or 'hovered on' for when you place the cursor on a text with an applied style.

There is a prop called "active" on dt-button that you can use to programatically set this state.

> Also there's one more issue which is that the 'muted' hover style matches the color of the background of the tool bar (at least in light mode), so I needed a slightly darker shade of gray.

Hmm yeah good point. I'll call design in for guidance on this one. @francisrupert @fede-dp @yorbi-dp 

We have this design of our "future message input" perhaps we want to use similar styling here, but I will let design make the decision on that. 
https://www.figma.com/file/oTY56LgQha8ljb4c8w4GtO/Dialpad-App-Components?node-id=9243%3A35258&mode=dev

Reply:
@emariotti-dialpad,

Which Product Designer are you working with on this feature? Please reach out to Dialtone Designers @fede-dp and @yorbi-dp for assistance beyond what I provided below.

Yeah, far too many base colors and CSS Utilities being applied when those colors are already accounted for in Button variants or should be using [surface, border, and foreground colors](https://dialtone.dialpad.com/design/colors/). For example, the toolbar's background color should be `var(--dt-color-surface-secondary)` (or the CSS Utility `d-bgc-secondary`).

The row of buttons should each be Muted variant and use the `active` prop for the selected one.

> ...the 'muted' hover style matches the color of the background of the tool bar...

Fortunately, the background color of `DTButton`'s Muted variant protects against this since it has a translucent background color, so this won't occur.

The attached screencast shows how that should render (just by me editing in inspector). 

https://github.com/dialpad/dialtone/assets/1165933/b3b3cd15-a62e-401a-abdf-e148dec8ec69